haskell/src/Ligo/Typescript.hs

```diff
@@ -315,6 +316,9 @@ mkTypesFor typename epType = case U.unwrapT epType of
   U.TBls12381G1 -> [TsType typename (TsAlias TsString)]
   U.TBls12381G2 -> [TsType typename (TsAlias TsString)]
   U.TNever -> [TsType typename (TsAlias TsVoid)]
+  U.TTicket a ->
+    let ticketFields = [("ticketer", toType U.TAddress), ("value", a), ("amount", toType U.TNat)]
+    in [TsType typename (TsInterface $ map (fst . mkTypesForField False) ticketFields)]
```

Comment:
Here you seem to be throwing away the `[TsDecl]` returned by  the `mkTypesForField`. This should be fine for the first and last elements for `ticketFields` as they are primitives. But for the second field, the type is `a`, which could require additional declarations. So if we throw the `[TsDecl]` away, the nested declarations for `a` won't be included. So I think something like the following might be required here (taken from the TPair branch of the same function.

```
        decls = (mkTypesForField False) <$> (indexEmptyFields allFields)
        recordFields = fst <$> decls
        thisType = TsType typename (TsInterface recordFields)
        thisDecls = snd <$> decls
        in thisType : (concat thisDecls)
```

```
        decls = (mkTypesForField False) <$> (indexEmptyFields allFields)
        recordFields = fst <$> decls
        thisType = TsType typename (TsInterface recordFields)
        thisDecls = snd <$> decls
        in thisType : (concat thisDecls)
```